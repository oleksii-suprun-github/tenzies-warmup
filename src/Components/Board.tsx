import React from 'react';
import { getRollDiceBtnLabel } from '../utils';
import Header from './UI/Header';
import Die from './UI/Die';
import RollDiceButton from './UI/RollDiceButton';
import { Dice, Difficulty } from 'types';

interface BoardProps {
  allDice: Dice[];
  difficulty: Difficulty;
  isGameWon: boolean;
  isGameStarted: boolean;
  rollDicesHandler: void;
  holdDieHandler: void;
}

const Board: React.FC<BoardProps> = ({
  allDice,
  difficulty,
  isGameWon,
  isGameStarted,
  rollDicesHandler,
  holdDieHandler,
}) => {
  const dice = allDice.length
    ? allDice.map((die) => (
        <Die
          key={die.id}
          holdDieHandler={() => holdDieHandler(die.id)}
          isHeld={die.isHeld}
          value={die.value}
        />
      ))
    : null;

  let buttonLabel = getRollDiceBtnLabel(isGameStarted, isGameWon);

  return (
    <>
      <Header title={`Match ${difficulty.value} dice`} />
      <section>
        {dice && <div id="dice-container">{dice}</div>}
        <RollDiceButton onClick={rollDicesHandler}>{buttonLabel}</RollDiceButton>
      </section>
    </>
  );
};
export default Board;
