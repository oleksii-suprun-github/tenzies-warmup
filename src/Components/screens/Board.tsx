import { FC } from 'react';
import Header from '../Header';
import Die from '../Die';
import RollDiceButton from '../RollDiceButton';

export interface BoardProps {
  allDice: Dice[];
  difficulty: GameDifficulty;
  isGameWon: boolean;
  isGameStarted: boolean;
  rollDicesHandler: () => void;
  holdDieHandler: Function;
}

const Board: FC<BoardProps> = ({ allDice, difficulty, rollDicesHandler, holdDieHandler }) => {
  const dice = allDice.map((die) => (
    <Die
      key={die.id}
      holdDieHandler={() => holdDieHandler(die.id)}
      isHeld={die.isHeld}
      value={die.value}
    />
  ));

  return (
    <>
      <Header>
        <h1
          data-testid="board-headline"
          className="mb-12 mt-12 text-4xl font-extrabold leading-none"
        >{`Match ${difficulty.value} dice`}</h1>
      </Header>
      <section className="justify-center; flex w-full flex-col items-center">
        {dice && (
          <div
            id="dice-container"
            className="mb-[50px] mt-[25px] grid gap-[25px] xs:grid-cols-[repeat(2,1fr)] sm-md:grid-cols-[repeat(3,1fr)] md:grid-cols-[repeat(5,1fr)]"
          >
            {dice}
          </div>
        )}
        <RollDiceButton onClick={rollDicesHandler}>Roll</RollDiceButton>
      </section>
    </>
  );
};
export default Board;
