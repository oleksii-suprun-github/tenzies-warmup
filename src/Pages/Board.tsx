import React from 'react';
import { getRollDiceBtnLabel } from '../utils';
import Header from '../components/Header';
import Die from '../components/Die';
import RollDiceButton from '../components/RollDiceButton';
import { Dice, Difficulty } from 'types';

interface BoardProps {
  allDice: Dice[];
  difficulty: Difficulty;
  isGameWon: boolean;
  isGameStarted: boolean;
  rollDicesHandler: () => void;
  holdDieHandler: Function;
}

const Board: React.FC<BoardProps> = ({
  allDice,
  difficulty,
  isGameWon,
  isGameStarted,
  rollDicesHandler,
  holdDieHandler,
}) => {
  const dice = allDice.length
    ? allDice.map((die) => (
        <Die
          key={die.id}
          holdDieHandler={() => holdDieHandler(die.id)}
          isHeld={die.isHeld}
          value={die.value}
        />
      ))
    : null;

  let buttonLabel = getRollDiceBtnLabel(isGameStarted, isGameWon);

  return (
    <>
      <Header>
        <h1>{`Match ${difficulty.value} dice`}</h1>
      </Header>
      <section>
        {dice && <div id="dice-container">{dice}</div>}
        <RollDiceButton onClick={rollDicesHandler}>{buttonLabel}</RollDiceButton>
      </section>
    </>
  );
};
export default Board;
