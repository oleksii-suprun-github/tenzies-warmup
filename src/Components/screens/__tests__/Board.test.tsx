import { render, screen } from '@testing-library/react';
import Board, { BoardProps } from '../Board';
import BoardMockProps from './__mocks__/BoardProps.json';

describe('Board', () => {
  it('should render a component with default props', () => {
    const { container } = render(<Board {...(BoardMockProps as BoardProps)} />);
    expect(container).toMatchSnapshot();
  });
  it('should render a correct amount of dice', () => {
    render(<Board {...(BoardMockProps as BoardProps)} />);
    const headlineText = screen.getByTestId('board-headline');
    expect(headlineText).toHaveTextContent('10');
  });
});
