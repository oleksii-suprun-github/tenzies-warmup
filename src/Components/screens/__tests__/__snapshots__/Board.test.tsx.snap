// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board should render a component with default props 1`] = `
<div>
  <header
    class="text-center"
    id="header"
  >
    <h1
      class="mb-12 mt-12 text-4xl font-extrabold leading-none"
      data-testid="board-headline"
    >
      Match 10 dice
    </h1>
  </header>
  <section
    class="justify-center; flex w-full flex-col items-center"
  >
    <div
      class="mb-[50px] mt-[25px] grid gap-[25px] xs:grid-cols-[repeat(2,1fr)] sm-md:grid-cols-[repeat(3,1fr)] md:grid-cols-[repeat(5,1fr)]"
      id="dice-container"
    >
      <button
        class="flex h-[100px] w-[100px] cursor-pointer flex-wrap items-center justify-center gap-[22px] self-center justify-self-center rounded-[10px] border-0 bg-main-die font-[bold] text-[42px] shadow-[0_10px_10px_rgba(0,0,0,0.1)] "
        data-testid="die"
      >
        <div
          class="h-2.5 w-2.5 rounded-[50%] bg-black pips-1 pip-index-1 "
        />
      </button>
      <button
        class="flex h-[100px] w-[100px] cursor-pointer flex-wrap items-center justify-center gap-[22px] self-center justify-self-center rounded-[10px] border-0 bg-main-die font-[bold] text-[42px] shadow-[0_10px_10px_rgba(0,0,0,0.1)] "
        data-testid="die"
      >
        <div
          class="h-2.5 w-2.5 rounded-[50%] bg-black pips-4 pip-index-1 "
        />
        <div
          class="h-2.5 w-2.5 rounded-[50%] bg-black pips-4 pip-index-2 bottom-[-61px] left-[-32px]"
        />
        <div
          class="h-2.5 w-2.5 rounded-[50%] bg-black pips-4 pip-index-3 "
        />
        <div
          class="h-2.5 w-2.5 rounded-[50%] bg-black pips-4 pip-index-4 right-[-31px]"
        />
      </button>
      <button
        class="flex h-[100px] w-[100px] cursor-pointer flex-wrap items-center justify-center gap-[22px] self-center justify-self-center rounded-[10px] border-0 bg-main-die font-[bold] text-[42px] shadow-[0_10px_10px_rgba(0,0,0,0.1)] "
        data-testid="die"
      >
        <div
          class="h-2.5 w-2.5 rounded-[50%] bg-black pips-2 pip-index-1 "
        />
        <div
          class="h-2.5 w-2.5 rounded-[50%] bg-black pips-2 pip-index-2 "
        />
      </button>
      <button
        class="flex h-[100px] w-[100px] cursor-pointer flex-wrap items-center justify-center gap-[22px] self-center justify-self-center rounded-[10px] border-0 bg-main-die font-[bold] text-[42px] shadow-[0_10px_10px_rgba(0,0,0,0.1)] "
        data-testid="die"
      >
        <div
          class="h-2.5 w-2.5 rounded-[50%] bg-black pips-2 pip-index-1 "
        />
        <div
          class="h-2.5 w-2.5 rounded-[50%] bg-black pips-2 pip-index-2 "
        />
      </button>
      <button
        class="flex h-[100px] w-[100px] cursor-pointer flex-wrap items-center justify-center gap-[22px] self-center justify-self-center rounded-[10px] border-0 bg-main-die font-[bold] text-[42px] shadow-[0_10px_10px_rgba(0,0,0,0.1)] "
        data-testid="die"
      >
        <div
          class="h-2.5 w-2.5 rounded-[50%] bg-black pips-3 pip-index-1 "
        />
        <div
          class="h-2.5 w-2.5 rounded-[50%] bg-black pips-3 pip-index-2 "
        />
        <div
          class="h-2.5 w-2.5 rounded-[50%] bg-black pips-3 pip-index-3 "
        />
      </button>
      <button
        class="flex h-[100px] w-[100px] cursor-pointer flex-wrap items-center justify-center gap-[22px] self-center justify-self-center rounded-[10px] border-0 bg-main-die font-[bold] text-[42px] shadow-[0_10px_10px_rgba(0,0,0,0.1)] "
        data-testid="die"
      >
        <div
          class="h-2.5 w-2.5 rounded-[50%] bg-black pips-3 pip-index-1 "
        />
        <div
          class="h-2.5 w-2.5 rounded-[50%] bg-black pips-3 pip-index-2 "
        />
        <div
          class="h-2.5 w-2.5 rounded-[50%] bg-black pips-3 pip-index-3 "
        />
      </button>
      <button
        class="flex h-[100px] w-[100px] cursor-pointer flex-wrap items-center justify-center gap-[22px] self-center justify-self-center rounded-[10px] border-0 bg-main-die font-[bold] text-[42px] shadow-[0_10px_10px_rgba(0,0,0,0.1)] "
        data-testid="die"
      >
        <div
          class="h-2.5 w-2.5 rounded-[50%] bg-black pips-2 pip-index-1 "
        />
        <div
          class="h-2.5 w-2.5 rounded-[50%] bg-black pips-2 pip-index-2 "
        />
      </button>
      <button
        class="flex h-[100px] w-[100px] cursor-pointer flex-wrap items-center justify-center gap-[22px] self-center justify-self-center rounded-[10px] border-0 bg-main-die font-[bold] text-[42px] shadow-[0_10px_10px_rgba(0,0,0,0.1)] "
        data-testid="die"
      >
        <div
          class="h-2.5 w-2.5 rounded-[50%] bg-black pips-3 pip-index-1 "
        />
        <div
          class="h-2.5 w-2.5 rounded-[50%] bg-black pips-3 pip-index-2 "
        />
        <div
          class="h-2.5 w-2.5 rounded-[50%] bg-black pips-3 pip-index-3 "
        />
      </button>
      <button
        class="flex h-[100px] w-[100px] cursor-pointer flex-wrap items-center justify-center gap-[22px] self-center justify-self-center rounded-[10px] border-0 bg-main-die font-[bold] text-[42px] shadow-[0_10px_10px_rgba(0,0,0,0.1)] "
        data-testid="die"
      >
        <div
          class="h-2.5 w-2.5 rounded-[50%] bg-black pips-3 pip-index-1 "
        />
        <div
          class="h-2.5 w-2.5 rounded-[50%] bg-black pips-3 pip-index-2 "
        />
        <div
          class="h-2.5 w-2.5 rounded-[50%] bg-black pips-3 pip-index-3 "
        />
      </button>
      <button
        class="flex h-[100px] w-[100px] cursor-pointer flex-wrap items-center justify-center gap-[22px] self-center justify-self-center rounded-[10px] border-0 bg-main-die font-[bold] text-[42px] shadow-[0_10px_10px_rgba(0,0,0,0.1)] "
        data-testid="die"
      >
        <div
          class="h-2.5 w-2.5 rounded-[50%] bg-black pips-6 pip-index-1 "
        />
        <div
          class="h-2.5 w-2.5 rounded-[50%] bg-black pips-6 pip-index-2 "
        />
        <div
          class="h-2.5 w-2.5 rounded-[50%] bg-black pips-6 pip-index-3 "
        />
        <div
          class="h-2.5 w-2.5 rounded-[50%] bg-black pips-6 pip-index-4 "
        />
        <div
          class="h-2.5 w-2.5 rounded-[50%] bg-black pips-6 pip-index-5 "
        />
        <div
          class="h-2.5 w-2.5 rounded-[50%] bg-black pips-6 pip-index-6 "
        />
      </button>
    </div>
    <button
      class="btn btn-primary mx-auto my-[25px] mt-[50px] w-fit min-w-32 bg-main-button text-2xl text-main-die"
      data-testid="roll-dice-button"
    >
      Roll
    </button>
  </section>
</div>
`;
